samples/coop_synchronization/sample.yaml

```diff
@@ -0,0 +1,41 @@
+sample:
+  description: A simple application that demonstrates even more basic sanity of the kernel
+    compared to samples/synchronization. The main variant is the polled_uart which shouldn't
+    go through any interrupt driven context switch in a no-systick scenario.
+  name: Coop Synchronization Sample
+tests:
+  sample.kernel.coop_synchronization.polled_uart:
+    # Platforms below do not implement a polled uart driver therefore they dont build
+    platform_exclude:
+      - rpi_pico
+      - rpi_pico_w
+      - adafruit_kb2040
+      - m2gl025_miv
+      - lpcxpresso51u68
+      - lpcxpresso54114_m4
+      - lpcxpresso55s06
+      - lpcxpresso55s16
+      - lpcxpresso55s28
+      - lpcxpresso55s36
+      - lpcxpresso55s69_cpu0
+      - lpcxpresso55s69_ns
+      - s32z270dc2_rtu0_r52
+      - s32z270dc2_rtu1_r52
+      - sparkfun_pro_micro_rp2040
+    tags: synchronization
+    extra_configs:
+      - CONFIG_UART_INTERRUPT_DRIVEN=n
+    harness: console
+    harness_config:
+      type: multi_line
+      regex:
+        - "thread_a: Hello World from cpu (\\d+) on (\\w+)! counter = (\\d+)"
+        - "thread_b: Hello World from cpu (\\d+) on (\\w+)! counter = -(\\d+)"
+  sample.kernel.coop_synchronization.interrupt_driven_uart:
+    tags: synchronization
+    harness: console
+    harness_config:
+      type: multi_line
+      regex:
+        - "thread_a: Hello World from cpu (\\d+) on (\\w+)! counter = (\\d+)"
+        - "thread_b: Hello World from cpu (\\d+) on (\\w+)! counter = -(\\d+)"
```

Comment:
We can move common parts to the `common` section:

```suggestion
sample:
  description: A simple application that demonstrates even more basic sanity of the kernel
    compared to samples/synchronization. The main variant is the polled_uart which shouldn't
    go through any interrupt driven context switch in a no-systick scenario.
  name: Coop Synchronization Sample
common:
  harness: console
  harness_config:
    type: multi_line
    regex:
      - "thread_a: Hello World from cpu (\\d+) on (\\w+)! counter = (\\d+)"
      - "thread_b: Hello World from cpu (\\d+) on (\\w+)! counter = -(\\d+)"
tests:
  sample.kernel.coop_synchronization.polled_uart:
    # Platforms below do not implement a polled uart driver therefore they dont build
    platform_exclude:
      - rpi_pico
      - rpi_pico_w
      - adafruit_kb2040
      - m2gl025_miv
      - lpcxpresso51u68
      - lpcxpresso54114_m4
      - lpcxpresso55s06
      - lpcxpresso55s16
      - lpcxpresso55s28
      - lpcxpresso55s36
      - lpcxpresso55s69_cpu0
      - lpcxpresso55s69_ns
      - s32z270dc2_rtu0_r52
      - s32z270dc2_rtu1_r52
      - sparkfun_pro_micro_rp2040
    tags: synchronization
    extra_configs:
      - CONFIG_UART_INTERRUPT_DRIVEN=n
  sample.kernel.coop_synchronization.interrupt_driven_uart:
    tags: synchronization
```

Reply:
Committed then fixed-up.